book/templates.rst

```diff
@@ -1,200 +1,721 @@
+
+.. Caution::
+
+    A template is shown in the dropdown only if both the XML and the Twig file
+    exist! If it is not, double-check the directories
+    ``app/Resources/templates/pages`` and ``app/Resources/views/templates``.
+
+The name displayed in the dropdown is configured in the ``<meta>`` section of
+the XML:
 
 .. code-block:: xml
 
+    <!-- app/Resources/templates/pages/default.xml -->
     <?xml version="1.0" ?>
     <template xmlns="http://schemas.sulu.io/template/template"
               xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
               xsi:schemaLocation="http://schemas.sulu.io/template/template http://schemas.sulu.io/template/template-1.0.xsd">
-        <key>default</key>
-
-        <view>ClientWebsiteBundle:templates:default</view>
-        <controller>SuluWebsiteBundle:Default:index</controller>
-        <cacheLifetime>2400</cacheLifetime>
+        <!-- ... -->
 
         <meta>
             <title lang="en">Default</title>
         </meta>
 
-        <properties>
-            <property name="title" type="text_line" mandatory="true">
-                <meta>
-                    <title lang="en">Title</title>
-                </meta>
+        <!-- ... -->
+    </template>
 
-                <tag name="sulu.rlp.part"/>
-            </property>
+You can customize the text by changing this property.
 
-            <property name="url" type="resource_locator" mandatory="true">
-                <meta>
-                    <title lang="en">Resourcelocator</title>
-                </meta>
+Creating a Custom Template
+--------------------------
 
-                <tag name="sulu.rlp"/>
-            </property>
+In your projects, you will need several templates for different parts of your
+website. The easiest way is to copy and adjust the default template.
+
+The first thing you need to adjust is the ``<key>``. This is the unique
+identifier of the template:
+
+.. code-block:: xml
+
+    <!-- app/Resources/templates/pages/event.xml -->
+    <?xml version="1.0" ?>
+    <template xmlns="http://schemas.sulu.io/template/template"
+              xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:schemaLocation="http://schemas.sulu.io/template/template http://schemas.sulu.io/template/template-1.0.xsd">
 
-            <property name="images" type="media_selection">
+        <key>event</key>
+
+        <!-- ... -->
+    </template>
+
+The second thing you have to customize is the ``<view>``. This element stores
+the Twig file that is used to render the template:
+
+.. code-block:: xml
+
+    <!-- app/Resources/templates/pages/event.xml -->
+    <?xml version="1.0" ?>
+    <template xmlns="http://schemas.sulu.io/template/template"
+              xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:schemaLocation="http://schemas.sulu.io/template/template http://schemas.sulu.io/template/template-1.0.xsd">
+        <!-- ... -->
+
+        <view>ClientWebsiteBundle:templates:event</view>
```

Comment:
in symfony you would have to write the extension. it sulu specific that this will be appended!